signer/vault/config.go

```diff
@@ -0,0 +1,36 @@
+package vault
+
+import (
+	"time"
+
+	vault "github.com/hashicorp/vault/api"
+
+	"github.com/dexidp/dex/pkg/log"
+	"github.com/dexidp/dex/signer"
+)
+
+type Config struct {
+	Address string `json:"address" yaml:"address"`
+	// TransitMount is the path to transit storage engine
+	TransitMount string `json:"mount" yaml:"mount"`
+	KeyName      string `json:"key" yaml:"key"`
+}
+
+func (c Config) Open(logger log.Logger, tokenValid time.Duration, rotationPeriod time.Duration) (signer.Signer, error) {
+	client, err := vault.NewClient(&vault.Config{
+		AgentAddress: c.Address,
```

Comment:
This means that using this feature requires Vault Agent, right?

Reply:
It's not actually requiring Vault agent - both Vault and Vault agent has the same API. The automated test also run on Vault directly.

However, there's no authentication support in this PR - at best you can supply VAULT_TOKEN which Vault SDK read from. Therefore for most users I'd imagine they'd need to use Vault agent to provide an authenticating proxy.

I'm not sure why upstream use this parameter name instead of just Vault server, but using it would affirm the intention that authentication won't be supported in Dex.

Reply:
Thanks for the explanation. There are some Vault clients that can auto-auth to Vault based on some underlying auth mechanism like AWS/GCP IAM, Kubernetes Service accounts like: https://github.com/banzaicloud/bank-vaults/blob/master/pkg/sdk/vault/client.go#L177